src/App.tsx

```diff
@@ -1,61 +1,105 @@
 import './App.scss';
+import { TodoList } from './components/TodoList';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import React, { useState } from 'react';
+import { Todos, TodoWithUser, User } from './types';
+
+const todosWithUsers = (todos: Todos[], users: User[]): TodoWithUser[] => {
+  return todos.map(todo => {
+    const match = users.find(user => user.id === todo.userId);
+
+    return {
+      ...todo,
+      name: match?.name || '',
+      username: match?.username || '',
+      email: match?.email || '',
+    };
+  });
+};
 
 export const App = () => {
+  const [title, setTitle] = useState<string>('');
+  const [onSubmit, setOnSubmit] = useState<boolean>(false);
```

Comment:
The state variable `onSubmit` is misleadingly named because it conflicts with the `onSubmit` event handler. Consider renaming it to something like `isFormSubmitted` to better reflect its purpose.